firmware/library/L1_Drivers/button_driver.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/pin_configure.hpp"
+
+class ButtonInterface {
+ public:
+         virtual bool getButtonState()      = 0;
+         virtual bool released()            = 0;
+         virtual bool pressed()             = 0;
+         virtual void defaultButtonSignal() = 0;
+         virtual void invertButtonSignal()  = 0;
+};
```

Comment:
Class methods names should have their first letter capitalized as per the Google C++ style guide.
https://google.github.io/styleguide/cppguide.html#General_Naming_Rules